[CS2113-F12-2] EquipmentManager

shunyao643 opened the pull request:
EquipmentManager is a Command Line Interface application to help with keeping track of equipment details (e.g. current user, quantity, cost, warranty duration, purchase date) for the AV club. It provides a clean and fast way to manage the inventory as compared to “traditional” methods such as an excel spreadsheet.

yuhaochua committed:
Shift tests for prepareModification to ModificationCommandTest from UpdateCommandTest

shunyao643 committed:
Merge branch 'master' into branch-implementUiClass

shunyao643 committed:
Merge pull request #58 from yuhaochua/branch-implementUiClass

Branch implement UI class

shunyao643 committed:
Merge branch 'master' into branch-unorderedTags

shunyao643 committed:
Merge pull request #57 from yuhaochua/branch-unorderedTags

Branch unordered tags

shunyao643 committed:
Update all classes for Java CI checks

shunyao643 committed:
Fix remaining pending Java CI errors

yuhaochua committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F12-2/tp

* 'master' of https://github.com/AY2122S2-CS2113-F12-2/tp:
  Fix remaining pending Java CI errors
  Update all classes for Java CI checks
  Shift tests for prepareModification to ModificationCommandTest from UpdateCommandTest
  Update code based on comments given by @shunyao643 for pull request 57
  Update creatioin of Parser object to be outside of do while loop
  Update welcome message to tell users to enter help if they need help.
  Update Javadocs for HelpCommand.execute()
  Add HelpCommand to show available commands to users.
  Add UI class to have UI displaying messages to users.
  Add JUnit test for new method checkAttributes() in AddCommand
  Rename DuplicateSerialNumber to DuplicateSerialNumberException
  Update AddCommand to support any order of strings given to add new equipment
  Add logic to check for successful update in UpdateCommand
  Add exception
  Add Storage class

shunyao643 committed:
Merge branch 'master' into parser

yuhaochua committed:
Fix Java CI errors from test files

yuhaochua committed:
Merge pull request #60 from yuhaochua/master

Fix Java CI errors arising from test files

shunyao643 committed:
Add support for parsing command word regardless of case

Additionally, improved documentation for splitCommandTerm

shunyao643 committed:
Switch prepareAdd to extractArguments for Parser

@yuhaochua, please check if this is the expected behaviour from your end!

shunyao643 committed:
Add helper method to set argument tags to lowercase

Work in progress

shunyao643 committed:
Reformat Regex for improved readability

yuhaochua committed:
Remove I/O redirection checks from github gradle build

yuhaochua committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F12-2/tp

* 'master' of https://github.com/AY2122S2-CS2113-F12-2/tp:

shunyao643 committed:
Include regex101.com debug link for easy maintainence

shunyao643 committed:
Merge branch 'master' into parser

shunyao643 committed:
Replace magic constants in Parser's prepareView and prepareDelete with named elements

shunyao643 committed:
Deprecate prepareAdd entirely

Rationale: extractArguments can handle all instances of prepareAdd

shunyao643 committed:
Return type enumeration in all caps

resolves #62

shunyao643 committed:
Attempt to change String in value of Pair to generic wildcard class

shunyao643 committed:
Add documentation for architecture,
parser and acknowledgements

yuhaochua committed:
Merge pull request #61 from shunyao643/parser

Support parsing of capitalised commands, enumeration and argument tags

yuhaochua committed:
Merge pull request #64 from shunyao643/developer-guide

Developer Guide

yuhaochua committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F12-2/tp

* 'master' of https://github.com/AY2122S2-CS2113-F12-2/tp:
  Add documentation for architecture, parser and acknowledgements
  Return type enumeration in all caps
  Deprecate prepareAdd entirely
  Replace magic constants in Parser's prepareView and prepareDelete with named elements
  Include regex101.com debug link for easy maintainence
  Reformat Regex for improved readability
  Add helper method to set argument tags to lowercase
  Switch prepareAdd to extractArguments for Parser
  Add support for parsing command word regardless of case
  Implement full equals method for IncorrectCommand for JUnit test

yuhaochua committed:
Fix bug where app ccrashes

Fixes #65

yuhaochua committed:
Minor edit to make output shown to user more presentable when using AddCommand

shunyao643 committed:
Add support for lowercase argument for list command

Test case written as well with ListCommand equals implementation

yuhaochua committed:
Add JavaDocs to methods in UpdateCommand.

yuhaochua committed:
Update Developer Guide and add images used.

yuhaochua committed:
Add UML diagram visual paradigm file

yuhaochua committed:
Reorder sequence in design and implementation

yuhaochua committed:
Merge pull request #67 from yuhaochua/branch-AddToDevGuide

Branch add to dev guide

yuhaochua committed:
Merge pull request #66 from shunyao643/parser

Add support for lowercase argument for list command

yuhaochua committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F12-2/tp

* 'master' of https://github.com/AY2122S2-CS2113-F12-2/tp:
  Reorder sequence in design and implementation
  Add UML diagram visual paradigm file
  Update Developer Guide and add images used.
  Add JavaDocs to methods in UpdateCommand.
  Add support for lowercase argument for list command
  Minor edit to make output shown to user more presentable when using AddCommand
  Fix bug where app ccrashes

Bryan-BC committed:
Add JavaDocs

Bryan-BC committed:
Update DeveloperGuide

Bryan-BC committed:
Merge pull request #68 from Bryan-BC/developerGuideAndJavaDocs

Developer guide and java docs

Bryan-BC committed:
Fix gradle

Bryan-BC committed:
Merge pull request #69 from Bryan-BC/developerGuideAndJavaDocs

Fix gradle

yuhaochua committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F12-2/tp

* 'master' of https://github.com/AY2122S2-CS2113-F12-2/tp:
  Fix gradle
  Update DeveloperGuide
  Add JavaDocs

yuhaochua committed:
Update DeveloperGuide to apply theme

shunyao643 committed:
Update Pair class to support parsing of Double cost and EquipmentType

shunyao643 committed:
Merge branch 'master' into pair-update

Bryan-BC committed:
Merge pull request #63 from shunyao643/pair-update

Pair update

shunyao643 committed:
Add Parser support for new Save command

closes #70

shunyao643 committed:
Fix Gradle CI/CD check for line exceeding 120 characters

yuhaochua committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F12-2/tp

* 'master' of https://github.com/AY2122S2-CS2113-F12-2/tp:
  Fix Gradle CI/CD check for line exceeding 120 characters
  Update Pair class to support parsing of Double cost and EquipmentType
  Attempt to change String in value of Pair to generic wildcard class

yuhaochua committed:
Fix developer guide table of content

yuhaochua committed:
Update user guide, by yuhao

yuhaochua committed:
Update user guide formatting

yuhaochua committed:
Merge pull request #71 from shunyao643/parser

Add Parser support for new Save command

yuhaochua committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F12-2/tp

* 'master' of https://github.com/AY2122S2-CS2113-F12-2/tp:
  Add Parser support for new Save command

yuhaochua committed:
Add SaveCommand

yuhaochua committed:
Fix error message which pops up after user inputs bye.

Fixes #74

yuhaochua committed:
Fix checkstyle error

yuhaochua committed:
Fix gradle error

yta-awu commented on docs/DeveloperGuide.md:
```diff
@@ -1,14 +1,123 @@
+- [User stories](#User-Stories)
+- [NonFunctional requirements](#NonFunctional Requirements)
+- [Glossary](#Glossary)
+- [Instructions for manual testing](#Instructions-for-manual-testing)
```
The link for Table of Content does not work.

yta-awu commented on docs/DeveloperGuide.md:
```diff
@@ -1,14 +1,123 @@
 
+This section describes some noteworthy details on how certain features are designed and implemented.
+
+### Architecture
```
Probably it's better to have an Architecture Diagram to aid understanding.

yta-awu commented on docs/DeveloperGuide.md:
```diff
@@ -1,14 +1,123 @@
+**`Parser`** is the first filter for text inputs read to the user. 
+As alluded to prior, one major consideration was to build it in a manner that can parse text input as effectively as possible. In considering text input, we divided the parsing into the following segments:
+
+#### `commandWord` `argumentTag/argumentValue` `argumentTag/argumentValue` `...`
```
Perhaps there's some mistyping for the last one? It's the same as the second one.

HansHengGit replied:
Consider giving an example usage for the add, delete and update operation.

yta-awu commented on docs/DeveloperGuide.md:
```diff
@@ -1,14 +1,123 @@
+
+**2. Split Arguments**
+
+For simpler command words such as **`delete`** and **`check`**, only one pair of **`argumentType`** and **`argumentValue`** were required. 
+As such, these will be passed in directly to their **`Command`** classes without a **`argumentType`** pair.
+
+More complex command words such as **`add`** and **`update`** necessitated multiple arguments.
+To implement this while ensuring that multi-word strings are acceptable input, **`extractArguments`** was implemented. 
+Without specifically explaining the main regular expression ([details here](https://regex101.com/r/gwjHWD/3)), the approach sought to match **`argumentTag/argumentValue`** pairs with a positive lookahead. 
+The final argument pair will then be extracted using a separate regex. 
+Together, this ensured that all argument pairs can be effectively parsed and dispatched to each **`Command`** class.
+
+Throughout the `Parser` implementation, exceptions were used to return `IncorrectCommand` classes that can be used to pass error messages to the user. These will be discussed in the following segments.
```
Do you try to emphasize anything through bolded code? Now some are bolded, some are not. For example, `Parser` in line 73 and **`Command`**.  However, I didn't see any difference in important level.

shunyao643 replied:
Cheers for pointing this out! Have fixed it.

yta-awu reviewed (commented):
Overall looks good. If you can include more diagrams, it will help the readers' understanding.

yta-awu commented on docs/DeveloperGuide.md:
```diff
@@ -1,14 +1,123 @@
+
+The *Sequence Diagram* below shows how the objects/classes interact with each other in the case of an update command being issued by the user.
+
+![sequenceDiagram](images/sequenceDiagram.png)
```
For alternative paths, perhaps it is an optional path. If my understanding is correct, you don't have an else statement in your code.

HansHengGit replied:
![Screenshot 2022-03-31 100443](https://user-images.githubusercontent.com/97375116/160961805-c3a6db9b-bcac-4255-b1a8-8eabcf91d1d2.png)
Should the start of Parser activation bar be before the method is called?
Should the return of the Parser class be at the end of the activation bar?

HansHengGit replied:
![Screenshot 2022-03-31 101801](https://user-images.githubusercontent.com/97375116/160963009-0be903f7-f751-4a78-b0bd-16baed1e9b6c.png)
Should there be a return of control after calling prepareModification()?

HansHengGit replied:
![Screenshot 2022-03-31 104543](https://user-images.githubusercontent.com/97375116/160965911-f09b1ae0-b936-487e-8805-d42501a6953f.png)
Should there be a deactivation in the activation bar after returning from parser() call.

FaliciaOng replied:
![image](https://user-images.githubusercontent.com/69444353/160970731-39422b3c-3561-4269-bcd3-3ee5ff5de2c5.png)

Is there no return arrow when execute command is being called?

HansHengGit commented on docs/DeveloperGuide.md:
```diff
@@ -1,14 +1,123 @@
+* **`deleteEquipment`** — Deletes the specified **`Equipment`**.
+
+The **`EquipmentManager`** is implemented as the main storage unit of all the equipments during the execution of the program.
+
```
Consider adding the steps of the implementation of EquipmentManager as per the Parser Component.

HansHengGit reviewed (commented):
Overall good work, but more components can be added to make your DG clearer, with the inclusion of more examples and more diagrams.

FaliciaOng commented on docs/DeveloperGuide.md:
```diff
@@ -1,14 +1,123 @@
+
+The *Sequence Diagram* below shows how the objects/classes interact with each other in the case of an update command being issued by the user.
+
+![sequenceDiagram](images/sequenceDiagram.png)
```
![image](https://user-images.githubusercontent.com/69444353/160969853-18131d14-0e41-4749-bcd5-a8497f429937.png)
Should the return arrow be at the end of the activation bar and the solid arrow be at the start of the activation bar?

WrinklyToad replied:
Maybe the calling arrow should touch the very top of the activation bar on the right, and the return arrow should come from the very bottom. This would signify that updateEquipment begins its lifecycle and isSuccessfulIUpdate ends it

FaliciaOng commented on docs/DeveloperGuide.md:
```diff
@@ -1,14 +1,123 @@
+
+### EquipmentManager
+
+The **`EquipmentManager`**, as the name suggests, manages all the equipment that the user has.
```
Consider using class or object diagram to illustrate how equipment manager works

FaliciaOng reviewed (commented):
Can consider adding more UML diagrams to help user understand the flow of the program

shunyao643 committed:
Merge pull request #75 from yuhaochua/branch-SaveCommand

SaveCommand and ByeCommand

shunyao643 committed:
Merge pull request #72 from yuhaochua/branch-updateUserGuide

Branch update user guide

shunyao643 committed:
Minor text edit for better user readability

shunyao643 committed:
Rename prepareView to prepareCheck in line with rest of the application's methods.

shunyao643 committed:
Test Markdown block

yuhaochua committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F12-2/tp

* 'master' of https://github.com/AY2122S2-CS2113-F12-2/tp:
  Rename prepareView to prepareCheck in line with rest of the application's methods.
  Minor text edit for better user readability
  Fix gradle error
  Fix checkstyle error
  Fix error message which pops up after user inputs bye.
  Add SaveCommand
  Update user guide formatting
  Update user guide, by yuhao
  Fix developer guide table of content

shunyao643 committed:
Update UserGuide for list, update and delete commands

shunyao643 committed:
Implement JUnit tests for delete function of app

shunyao643 committed:
Improve regex detection by stripping whitespace for one argument commands

shunyao643 committed:
Merge pull request #76 from shunyao643/user-guide

User guide: Update for list, update and delete commands

WrinklyToad commented on docs/DeveloperGuide.md:
```diff
@@ -1,14 +1,123 @@
+- **`EquipmentManager`** keeps track of actual **`Equipment`** instances created by our program.
+- **`Storage`** performs File I/O functions.
+
+### Parser
```
A visual diagram of Parser may make this section easier to understand

WrinklyToad commented on docs/DeveloperGuide.md:
```diff
@@ -1,14 +1,123 @@
+to the design and user interactions when it came to the text input to be parsed by the user. 
+As such, we referenced AddressBook to segment the program into **`Parser`**, **`Command`** and **`Equipment`** classes.
+
+**Main components of the architecture**
+
```
A class diagram could be appropriate here

WrinklyToad commented on docs/DeveloperGuide.md:
```diff
@@ -1,14 +1,123 @@
+### EquipmentManager
+
+The **`EquipmentManager`**, as the name suggests, manages all the equipment that the user has.
+When initialised, it creates a HashMap called **`equipmentList`** which is used to save the equipment with their serial numbers as their keys.
+The **`EquipmentManager`** has several methods which allow manipulation to said HashMap such as **`addEquipment`**, **`checkEquipment`**, **`listEquipment`**, **`updateEquipment`**, and **`deleteEquipment`**.
+These methods are used during the execution of each of the **`Command`** class.
+* **`addEquipment`** — Adds an **`Equipment`** to the **`equipmentList`**.
+* **`checkEquipment`** — Returns the **`Equipment`** in **`equipmentList`** that has the given **`itemName`**.
+* **`listEquipment`** — Returns an ArrayList of all the **`Equipment`** in **`equipmentList`**.
+* **`updateEquipment`** — Updates the specified **`Equipment`** with the corresponding updates given in the ArrayList of pairs.
+* **`deleteEquipment`** — Deletes the specified **`Equipment`**.
+
+The **`EquipmentManager`** is implemented as the main storage unit of all the equipments during the execution of the program.
```
Explanations of each of these features may be beneficial

WrinklyToad commented on docs/DeveloperGuide.md:
```diff
@@ -1,14 +1,123 @@
+
+![equipment1](images/equipment1.png)
+
+The *Sequence Diagram* below shows how the objects/classes interact with each other in the case of an update command being issued by the user.
```
If all the other commands interact in a similar or almost exact way to this, maybe it could be noted here. More diagrams may still be helpful, but it could be a good clarification

shunyao643 replied:
This was certainly what we were trying to achieve. Good call!

WrinklyToad commented on docs/DeveloperGuide.md:
```diff
@@ -1,14 +1,123 @@
+
+The *Sequence Diagram* below shows how the objects/classes interact with each other in the case of an update command being issued by the user.
+
+![sequenceDiagram](images/sequenceDiagram.png)
```
![image](https://user-images.githubusercontent.com/60244374/161183877-86f6dc9e-4594-4c8c-a624-2b489718aef4.png)

Shouldn't the calling and returning arrows touch the very tops and bottoms of the activation bars?

WrinklyToad replied:
![image](https://user-images.githubusercontent.com/60244374/161184109-0c93d1ce-d68e-437d-b389-03f7d33ec380.png)

Is Parser making a new command object here? If Parser is making a new command object, it may be the diagram should signify a constructor call (by having the calling arrow touch the "command: UpdateCommand" box and have an activation bar immediately following that.

Mick609 replied:
The sequence diagram is a bit too big (in resolution). Placing this diagram in the final submission would be hard to be viewed.

WrinklyToad reviewed (commented):
Overall, I think it may be the case your DG could just benefit from some more explanation and visuals. The update equipment feature is well explained and feels sufficiently in-depth. Perhaps this could be applied to all of the other features. There may be a few things you could fix in your updateEquipment sequence diagram, but they should be relatively quick fixes!

Also, the class structure could be easier to follow with visuals at the beginning. Along with the steps that describe the class interactions, some diagrams could be helpful.

Mick609 commented on docs/DeveloperGuide.md:
```diff
@@ -1,14 +1,123 @@
+to the design and user interactions when it came to the text input to be parsed by the user. 
+As such, we referenced AddressBook to segment the program into **`Parser`**, **`Command`** and **`Equipment`** classes.
+
+**Main components of the architecture**
```
As others suggested, class diagrams or an architecture diagram would be appreciated here.

Mick609 commented on docs/DeveloperGuide.md:
```diff
@@ -1,14 +1,123 @@
+The **`EquipmentManager`** is implemented as the main storage unit of all the equipments during the execution of the program.
+
+--------------------------------------------------------------------------------------------------------------------
 ## Product scope
```
Product scope is necessary for the DG. Please try to finish this part soon.

Mick609 commented on docs/DeveloperGuide.md:
```diff
@@ -1,14 +1,123 @@
 
 ## Acknowledgements
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
```
You could maybe remove this placeholder in the final version?

shunyao643 replied:
Noted, will do so!

Mick609 commented on docs/DeveloperGuide.md:
```diff
@@ -18,21 +127,25 @@
 
 |Version| As a ... | I want to ... | So that I can ...|
 |--------|----------|---------------|------------------|
 |v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
```
We can definitely have more user stories.

Mick609 reviewed (commented):
Adding more diagrams (like class diagrams and architecture diagram) would be better.

yuhaochua committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F12-2/tp

* 'master' of https://github.com/AY2122S2-CS2113-F12-2/tp:
  Update UserGuide for list, update and delete commands
  Test Markdown block

Bryan-BC committed:
Add bye, help, and save commands to UserGuide

Bryan-BC committed:
Merge pull request #79 from Bryan-BC/userGuide

Add bye, help, and save commands to UserGuide

Bryan-BC committed:
Merge pull request #78 from shunyao643/parser

Improve Regex detection by stripping whitespace for one-argument commands

shunyao643 committed:
Merge branch 'master' of https://github.com/shunyao643/tp into parser

shunyao643 committed:
Add test case for decimal cost

shunyao643 committed:
Merge branch 'master' of https://github.com/shunyao643/tp into parser

shunyao643 committed:
Add support for float costs

Bryan-BC committed:
Fix documentation bug

Bryan-BC committed:
Fix typo in HelpCommand

yuhaochua committed:
Merge pull request #112 from Bryan-BC/userGuide

Fix documentation bug

Bryan-BC committed:
Add autosave feature

Bryan-BC committed:
Merge pull request #111 from shunyao643/parser

Add support for decimal costs in Parser

shunyao643 committed:
Refactor code to move deprecated methods, variables to end of class

yuhaochua committed:
Merge pull request #113 from Bryan-BC/autoSave

Add autosave feature

yuhaochua committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F12-2/tp

* 'master' of https://github.com/AY2122S2-CS2113-F12-2/tp:
  Add autosave feature
  Fix typo in HelpCommand
  Fix documentation bug
  Add support for float costs
  Add test case for decimal cost
  Add bye, help, and save commands to UserGuide
  Improve regex detection by stripping whitespace for one argument commands
  Implement JUnit tests for delete function of app

shunyao643 committed:
Complete refactoring of internal references from View command to Check command

shunyao643 committed:
Improve documentation for modification regex

shunyao643 committed:
Refactor code to trim strings in parseCommand

shunyao643 committed:
Update javadoc for splitCommandTerm

shunyao643 committed:
Improve user feedback for failed command parsing

yuhaochua committed:
Fix typo with output of HelpCommand being executed

Closes #81

yuhaochua committed:
Update instructions in UserGuide on starting the application

Closes #80

yuhaochua committed:
Fix problem where wrong message occurs when there are missing attributes for AddCommand

Fixes #90

yuhaochua committed:
Update the error message to be more detailed when user has arguments squished together

Closes #92

yuhaochua committed:
Update UserGuide on supporting of cost up to $10 million

Closes #98

yuhaochua committed:
Fix UpdateCommand not updating purchasedDate and purchasedFrom properly

Fixes #102

yuhaochua committed:
Fix broken JUnit test for EquipmentManager

shunyao643 committed:
Refactor Modification commands to require arguments to have backticks enclosed

shunyao643 committed:
Merge pull request #115 from yuhaochua/branch-fixUpdatePdPf

Branch fix update pd pf

shunyao643 committed:
Update test cases to reflect shift trim() to parseCommand

shunyao643 committed:
Update prepareCheck, prepareDelete to align with new backtick syntax

shunyao643 committed:
Update regex101 link

shunyao643 committed:
Update command help tips to include backtick requirement

yuhaochua committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F12-2/tp

* 'master' of https://github.com/AY2122S2-CS2113-F12-2/tp:
  Fix broken JUnit test for EquipmentManager
  Fix UpdateCommand not updating purchasedDate and purchasedFrom properly

shunyao643 committed:
Update check command to allow any argument pair, but just one.

shunyao643 committed:
Refine delete command to accept a more narrow range of characters

Bryan-BC committed:
Merge pull request #114 from yuhaochua/branch-cosmeticUpdates

Branch cosmetic updates

yuhaochua committed:
Merge pull request #116 from shunyao643/parser

Parser Code Cleanup in preparation for shift to require backticks

yuhaochua committed:
Merge branch 'master' into parser-backtick

yuhaochua committed:
Merge pull request #117 from shunyao643/parser-backtick

Parser Backtick update to require backticks

yuhaochua committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F12-2/tp

* 'master' of https://github.com/AY2122S2-CS2113-F12-2/tp:
  Refine delete command to accept a more narrow range of characters
  Update check command to allow any argument pair, but just one.
  Update command help tips to include backtick requirement
  Update regex101 link
  Update prepareCheck, prepareDelete to align with new backtick syntax
  Update test cases to reflect shift trim() to parseCommand
  Refactor Modification commands to require arguments to have backticks enclosed
  Update UserGuide on supporting of cost up to $10 million
  Update the error message to be more detailed when user has arguments squished together
  Fix problem where wrong message occurs when there are missing attributes for AddCommand
  Update instructions in UserGuide on starting the application
  Fix typo with output of HelpCommand being executed
  Improve user feedback for failed command parsing
  Update javadoc for splitCommandTerm
  Refactor code to trim strings in parseCommand
  Improve documentation for modification regex
  Complete refactoring of internal references from View command to Check command
  Refactor code to move deprecated methods, variables to end of class

yuhaochua committed:
Fix broken JUnit test in ModificationCommandTest

yuhaochua committed:
Change CheckCommand#execute to pass a Pair to EquipmentManager#checkEquipment instead of a String.

yuhaochua committed:
Fix some broken JUnit tests

yuhaochua committed:
Add JUnit tests for DeleteCommand

yuhaochua committed:
Update messages to show backticks in examples of Commands usage

shunyao643 committed:
Unbold code to reduce emphasis, in line with rest of DG

shunyao643 committed:
Update regex101 link

shunyao643 committed:
Revert "Update regex101 link": should be in
parser-backtick instead

This reverts commit 11dce75b8ec4bdbb682b3fd98a389acf5d47cc55.

shunyao643 committed:
Attempt to center code

shunyao643 committed:
Attempt to center formatted code

shunyao643 committed:
Fix code formatting for Parsr

shunyao643 committed:
Test code formatting for Parser

shunyao643 committed:
Test code formatting for Parser

shunyao643 committed:
Test code formatting for Parser

shunyao643 committed:
Test code formatting for Parser

shunyao643 committed:
Test code formatting for Parser

shunyao643 committed:
Include backticks in DG

shunyao643 committed:
Merge branch 'master' into developer-guide

shunyao643 committed:
Rewrite Parser DG documentation for v2.1

shunyao643 committed:
Rewrite Update command to be more general for Command

shunyao643 committed:
Remove acknowledgements placeholder

shunyao643 committed:
Reword Design and Implementation for better flow

shunyao643 committed:
Draft product scope

shunyao643 committed:
Add prepareModification() to test case call

shunyao643 committed:
Merge pull request #122 from shunyao643/extractArguments-fix

Add prepareModification() to test case call

Bryan-BC committed:
Refactor checkEquipment to take Pair as parameter

yuhaochua committed:
Made changes according to @shunyao643 comments on pull request #119

yuhaochua committed:
Merge branch 'master' into branch-newDeleteCommand

yuhaochua committed:
Merge pull request #120 from shunyao643/developer-guide

Update Developer Guide

yuhaochua committed:
Update value to be used in Pair for CheckCommand

yuhaochua committed:
Merge pull request #118 from yuhaochua/branch-newCheckCommand

Branch new check command

yuhaochua committed:
Merge pull request #123 from Bryan-BC/checkEquipment

Refactor checkEquipment to take Pair as parameter

yuhaochua committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F12-2/tp

* 'master' of https://github.com/AY2122S2-CS2113-F12-2/tp: (22 commits)
  Update value to be used in Pair for CheckCommand
  Refactor checkEquipment to take Pair as parameter
  Add prepareModification() to test case call
  Draft product scope
  Reword Design and Implementation for better flow
  Remove acknowledgements placeholder
  Rewrite Update command to be more general for Command
  Rewrite Parser DG documentation for v2.1
  Include backticks in DG
  Test code formatting for Parser
  Test code formatting for Parser
  Test code formatting for Parser
  Test code formatting for Parser
  Test code formatting for Parser
  Fix code formatting for Parsr
  Attempt to center formatted code
  Attempt to center code
  Revert "Update regex101 link": should be in parser-backtick instead
  Update regex101 link
  Unbold code to reduce emphasis, in line with rest of DG
  ...

yuhaochua committed:
Fix broken JUnit test for CheckCommand

yuhaochua committed:
Modify EquipmentManger#checkEquipment to use `contains` instead of `equals` for some of the attributes
Fix CheckCommand#generateCheckPair not putting the appropriate class for Cost and EquipmentType into the Pair

shunyao643 committed:
Merge pull request #119 from yuhaochua/branch-newDeleteCommand

Branch new delete command

yuhaochua committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F12-2/tp

* 'master' of https://github.com/AY2122S2-CS2113-F12-2/tp:
  Made changes according to @shunyao643 comments on pull request #119
  Add JUnit tests for DeleteCommand
  Fix some broken JUnit tests

shunyao643 committed:
Fix checkstyle

shunyao643 committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F12-2/tp

shunyao643 committed:
Draft date implementation

shunyao643 committed:
Add LocalDate parsing to CheckCommand

yuhaochua committed:
Remove for loop in CheckCommand#generateCheckPair

yuhaochua committed:
Merge pull request #124 from shunyao643/date-implementation

Date implementation

yuhaochua committed:
Merge branch 'master' into branch-checkCommandLoop

yuhaochua committed:
Merge pull request #126 from yuhaochua/branch-checkCommandLoop

Remove for loop in CheckCommand#generateCheckPair

yuhaochua committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F12-2/tp

* 'master' of https://github.com/AY2122S2-CS2113-F12-2/tp:
  Remove for loop in CheckCommand#generateCheckPair
  Add LocalDate parsing to CheckCommand
  Draft date implementation
  Fix checkstyle

yuhaochua committed:
Update Project Portfolio Page

yuhaochua committed:
Minor edit to documentation in DeveloperGuide and UserGuide

yuhaochua committed:
Edit sublist style for review contributions

yuhaochua committed:
Update sublist

yuhaochua committed:
Fix references to Project Portfolio Pages

shunyao643 committed:
Merge pull request #127 from yuhaochua/branch-yuhaoPPP

Update Project Portfolio Page for yuhao

Bryan-BC committed:
Refactor UserGuide.md to include code blocks and update some command details.

Bryan-BC committed:
Refactor UserGuide.md to include code blocks and update some command details.

shunyao643 committed:
Update Shun Yao's PPP

Bryan-BC committed:
Update portfolio for bryanchristopher.md

shunyao643 committed:
Merge pull request #129 from Bryan-BC/Bryan-AboutUs

Update portfolio for bryanchristopher.md

yuhaochua committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F12-2/tp

* 'master' of https://github.com/AY2122S2-CS2113-F12-2/tp:
  Update portfolio for bryanchristopher.md
  Update Shun Yao's PPP
  Fix references to Project Portfolio Pages
  Update sublist
  Edit sublist style for review contributions
  Minor edit to documentation in DeveloperGuide and UserGuide
  Update Project Portfolio Page

yuhaochua committed:
Refactor error messages from ModificationCommand into Command

shunyao643 committed:
Rename Shun Yao's PPP in line with requirements

shunyao643 committed:
Merge pull request #130 from shunyao643/shunyao-PPP

Rename Shun Yao's PPP in line with requirements

Bryan-BC committed:
Add support for LocalDate in Storage

Bryan-BC committed:
Add support for LocalDate in Storage

Bryan-BC committed:
Merge branch 'storageTweaks' of https://github.com/Bryan-BC/tp into storageTweaks

Bryan-BC committed:
Merge branch 'userGuide' of https://github.com/Bryan-BC/tp into userGuide

Bryan-BC committed:
Fix Gradle and removed list TYPE from HelpCommand

Bryan-BC committed:
Remove * import

yuhaochua committed:
Merge pull request #128 from Bryan-BC/userGuide

Refactor UserGuide.md to include code blocks and update some command …

yuhaochua committed:
Merge pull request #131 from Bryan-BC/storageTweaks

Storage tweaks

yuhaochua committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F12-2/tp into branchDocumentations

* 'master' of https://github.com/AY2122S2-CS2113-F12-2/tp:
  Remove * import
  Fix Gradle and removed list TYPE from HelpCommand
  Refactor UserGuide.md to include code blocks and update some command details.
  Add support for LocalDate in Storage
  Add support for LocalDate in Storage
  Rename Shun Yao's PPP in line with requirements
  Refactor UserGuide.md to include code blocks and update some command details.

Bryan-BC committed:
Make key of Equipments to be lowercase

Bryan-BC committed:
Merge pull request #133 from Bryan-BC/equipmentManagerTweaks

Make key of Equipments to be lowercase

yuhaochua committed:
Update exception handling for CheckCommand

Fixes #134

yuhaochua committed:
Add instructions for manual testing to developer guide

yuhaochua committed:
Fix checkstyle

yuhaochua committed:
Remove implementation of listing equipment of specified type

yuhaochua committed:
Remove remaining traces of listing equipment with type

yuhaochua committed:
Update java docs

yuhaochua committed:
Add to glossary in DeveloperGuide

yuhaochua committed:
Add and update JUnit tests

yuhaochua committed:
Edit sequence diagram and update developer guide

yuhaochua committed:
Update glossary

yuhaochua committed:
Fix formatting issue with developer guide

shunyao643 committed:
Loosen restrictions for modification command regex

shunyao643 committed:
Deprecate support for list command with argument tags in Parser

shunyao643 committed:
Refactor Parser#splitCommandTerm with updated JavaDoc

shunyao643 committed:
Elaborate on DeveloperGuide.md's Design and Implementation

shunyao643 committed:
Elaborate on DeveloperGuide.md's Design and Implementation

shunyao643 committed:
Refactor constants in Parser.java

shunyao643 committed:
Remove deprecated code

shunyao643 committed:
Remove deprecated test case for parseCommand

shunyao643 committed:
Minor edit to Value Proposition

shunyao643 committed:
Add some user stories

Bryan-BC committed:
Add Storage in DG and change file name

yuhaochua committed:
Refactor personal portfolio page name to align with github name

yuhaochua committed:
Merge pull request #136 from shunyao643/parser

Loosen restrictions for modification command regex

yuhaochua committed:
Merge pull request #135 from yuhaochua/branchDocumentations

Branch documentations

shunyao643 committed:
Merge branch 'master' into documentations

shunyao643 committed:
Merge pull request #139 from Bryan-BC/documentations

Add Storage in DG and change file name

shunyao643 committed:
Merge branch 'master' into parser-updated-list

shunyao643 committed:
Merge pull request #137 from shunyao643/parser-updated-list

Deprecate list support in Parser

shunyao643 committed:
Merge branch 'master' into developer-guide

shunyao643 committed:
Standardise formatting for Developer Guide

shunyao643 committed:
Reformat Instructions for manual testing

shunyao643 committed:
Add Parser terminology to glossary

shunyao643 committed:
Update Parser regex documentation

shunyao643 committed:
Update User Stories table

shunyao643 committed:
Touchup rest of DeveloperGuide.md

shunyao643 committed:
Catch DateTimeParseException for Update command

shunyao643 committed:
Fix uncaught exceptions in Parser

shunyao643 committed:
Complete JavaDoc for Parser.java

shunyao643 committed:
Update shunyao643.md

yuhaochua committed:
Merge pull request #141 from shunyao643/parser

Exception Handling in Parser

shunyao643 committed:
Update enhancements implemented

yuhaochua committed:
Merge pull request #142 from shunyao643/shunyao-PPP

Update shunyao643.md

Bryan-BC committed:
Merge pull request #138 from shunyao643/developer-guide

Developer Guide: Update Product Scope, User Stories, Non-Functional Requirements

shunyao643 committed:
Update Parser to remove unnecessary exceptions in CheckCommand

shunyao643 committed:
Remove disabled test cases in ParserTest.java

yuhaochua committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F12-2/tp into branchDocumentations

* 'master' of https://github.com/AY2122S2-CS2113-F12-2/tp: (23 commits)
  Update enhancements implemented
  Update shunyao643.md
  Complete JavaDoc for Parser.java
  Fix uncaught exceptions in Parser
  Catch DateTimeParseException for Update command
  Touchup rest of DeveloperGuide.md
  Update User Stories table
  Update Parser regex documentation
  Add Parser terminology to glossary
  Reformat Instructions for manual testing
  Standardise formatting for Developer Guide
  Add Storage in DG and change file name
  Add some user stories
  Minor edit to Value Proposition
  Remove deprecated test case for parseCommand
  Remove deprecated code
  Refactor constants in Parser.java
  Elaborate on DeveloperGuide.md's Design and Implementation
  Elaborate on DeveloperGuide.md's Design and Implementation
  Refactor Parser#splitCommandTerm with updated JavaDoc
  ...

Bryan-BC committed:
Add minor details to DG and change portfolio part in AboutUs

Bryan-BC committed:
Add minor details to DG and change portfolio part in AboutUs

Bryan-BC committed:
Merge branch 'documentationTweaks' of https://github.com/Bryan-BC/tp into documentationTweaks

shunyao643 committed:
Add test case for updateCommand to catch incorrect Date exception

yuhaochua committed:
Minor edit to sequenceDiagram screenshot

yuhaochua committed:
Minor edit to sequence diagram

yuhaochua committed:
More edit to sequence diagram execute screenshot

yuhaochua committed:
Update visual paradigm files

shunyao643 committed:
Transfer new user stories as planned futures into UserGuide.md

shunyao643 committed:
Merge pull request #143 from Bryan-BC/documentationTweaks

Documentation tweaks

shunyao643 committed:
Merge pull request #145 from yuhaochua/branchDocumentations

Update sequence diagrams

shunyao643 committed:
Merge pull request #144 from shunyao643/master

Finalise remaining code

shunyao643 committed:
Update UserGuide and PPP to reflect v2.1 features

shunyao643 committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F12-2/tp

shunyao643 committed:
Add more informative error for $ parsing

shunyao643 committed:
Merge pull request #146 from shunyao643/master

Add more informative error for $ parsing

Bryan-BC committed:
Change file name to bryan-bc.md

Bryan-BC committed:
Merge pull request #147 from Bryan-BC/documentationTweaks

Change file name to bryan-bc.md

Bryan-BC committed:
Add README in tp/docs

Bryan-BC committed:
Merge pull request #148 from Bryan-BC/documentationTweaks

Add README in tp/docs
